lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -65,14 +65,40 @@ public static function get_theme_data( $deprecated = array(), $settings = array(
 		if ( ! $settings['with_supports'] ) {
 			return static::$theme;
 		}
+		
+		$editor_settings = array(
+			'disableCustomColors'    => get_theme_support( 'disable-custom-colors' ),
+			'disableCustomFontSizes' => get_theme_support( 'disable-custom-font-sizes' ),
+			'disableCustomGradients' => get_theme_support( 'disable-custom-gradients' ),
+			'disableLayoutStyles'    => get_theme_support( 'disable-layout-styles' ),
+			'enableCustomLineHeight' => get_theme_support( 'custom-line-height' ),
+			'enableCustomSpacing'    => get_theme_support( 'custom-spacing' ),
+			'enableCustomUnits'      => get_theme_support( 'custom-units' ),
+		);
+
+		// Theme settings.
+		$color_palette = current( (array) get_theme_support( 'editor-color-palette' ) );
+		if ( false !== $color_palette ) {
+			$editor_settings['colors'] = $color_palette;
+		}
+
+		$font_sizes = current( (array) get_theme_support( 'editor-font-sizes' ) );
+		if ( false !== $font_sizes ) {
+			$editor_settings['fontSizes'] = $font_sizes;
+		}
+
+		$gradient_presets = current( (array) get_theme_support( 'editor-gradient-presets' ) );
+		if ( false !== $gradient_presets ) {
+			$editor_settings['gradients'] = $gradient_presets;
+		}
```

Comment:
All of the new code here is now a bit of a duplicate code from what is already in `get_default_block_editor_settings()`. I think we can solve the problem in a more efficient way by enhancing that function.